replit_river/codegen/server.py

```diff
@@ -95,45 +95,56 @@ def message_decoder(
             )
     # oneof fields.
     for index, oneof in enumerate(m.oneof_decl):
-        chunks.extend(
-            [
-                f"  _{oneof.name} = d.get('{to_camel_case(oneof.name)}', {{}})",
-                f"  if _{oneof.name}:",
-                f"    match _{oneof.name}.get('$kind', None):",
-            ]
-        )
-        for field in oneofs[index]:
-            chunks.append(
-                f"        case '{to_camel_case(field.name)}':",
+        for fieldIndex, field in enumerate(oneofs[index]):
+            accessor_key = (
+                "d"
+                if field.label == FieldDescriptor.LABEL_OPTIONAL
+                else f"_{oneof.name}"
             )
+
+            # don't wrap optional fields in additional object
+            if field.label == FieldDescriptor.LABEL_OPTIONAL:
+                chunks.append(f"  if d.get('{to_camel_case(field.name)}') is not None:")
+            else:
```

Comment:
we don't do a lot of ternaries, and we already have a branch for it.

```suggestion
            accessor_key: string

            # don't wrap optional fields in additional object
            if field.label == FieldDescriptor.LABEL_OPTIONAL:
                accessor_key = "d"
                chunks.append(f"  if d.get('{to_camel_case(field.name)}') is not None:")
            else:          
                accessor_key = f"_{oneof.name}"
```

Reply:
oh nice didn't know the pythonic way of doing that. easy enough